### DISCUSSION TITLE
Searching and Removing nested element in JSON payload without explicit check

### DISCUSSION THREAD

**Discussion by akashvekariya** [schibsted/jslt#284] [Q&A] [answered]
The problem I'm trying to solve here is that, let's consider the given is my test payload
```{
  "order_id": "67890",
  "customer": {
    "name": "Jane Smith",
    "email": "janesmith@example.com",
    "shipping_address": {
      "street": "456 Maple Ave",
      "city": "Anytown",
      "state": "CA",
      "zip": "12345",
      "country": "US"
    }
  },
  "items": [
    {
      "name": "Product 1",
      "sku": "P123",
      "quantity": 2,
      "price": 19.99
    },
    {
      "name": "Product 2",
      "sku": "P456",
      "quantity": 1,
      "price": 29.99
    }
  ],
  "shipping_method": "Expedited",
  "payment_method": "PayPal",
  "total": 69.97
}
```
and I want to remove the field zip form .customer.shipping-address! Now I can solve this by simply doing * - zip inside of shipping-address, but that's in this data. What if the payload returns data in which the zip is nested inside .employee.shipping-address! 

This thing may not be used in real life scenario but in my organization we're making a template service that transforms one payload to another and does not pass any sensitive data to end user.

**Comment by larsga** [comment]
So you want to remove some specific field regardless of where in the JSON it appears?

You can do that by writing a recursive function that traverses the JSON and for any object it finds it traverses that object and skips any field with the given name.

**Comment by akashvekariya** [comment, marked as answer]
Thanks, I did it.

Here's the code.
```
def deleteByKey(jsonObj, searchTerm)
    if (is-object($jsonObj))
      {for ($jsonObj)
        .key:
        if(.key != $searchTerm) 
           if (is-object(.value) or is-array(.value))
              deleteByKey(.value, $searchTerm)
           else
              .value
      }
    else if (is-array($jsonObj))
       [for($jsonObj)
            deleteByKey(., $searchTerm)
       ]
 
deleteByKey(., "shipping_address")
```

Do you think I can optimize it or is there any test case issues? Do let me know.

**Reply by larsga** [reply]
This looks good to me. I don't think there's much you can do to optimize this. Test cases I'm afraid you'll have to write yourself.
